test/utils/wasm_engine.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "types.hpp"
+#include <memory>
+#include <optional>
+
+namespace fizzy::test
+{
+/// The stateful representation of a wasm engine for testing purposes.
+class WasmEngine
+{
+public:
+    struct Result
+    {
+        bool trapped = false;
+        std::optional<uint64_t> value;
+    };
+
+    virtual ~WasmEngine() noexcept = default;
+
+    /// Parses input bytes and creates internal wasm module out of it.
+    /// Consecutive invokes replaces the internal module with a new one.
```

Comment:
```suggestion
    /// Consecutive invokes replaces the internal module (but not the instance) with a new one.
```

Reply:
I guess we should say it also invalidates the instance. Some things will not work as expected.